lib/tools/apk-utils.js

```diff
@@ -325,32 +327,43 @@ apkUtilsMethods.installOrUpgrade = async function (apk, pkg = null, timeout = 60
     log.warn(`Cannot read the package name of ${apk}. Assuming correct app version is already installed`);
     return;
   }
+
   if (!await this.isAppInstalled(pkg)) {
     await this.install(apk, false, timeout);
     return;
   }
+
+  if (fullReinstall) {
```

Comment:
could shorten this a bit by putting the check and uninstall ahead of the `isAppInstalled` check, then relying on the install to happen in that block. NBD either way.

Reply:
yep, uninstallApk already contains the verification of whether the destination app is installed or not